src/CommunityToolkit/Xamarin.CommunityToolkit/Views/AvatarView/AvatarView.shared.cs

```diff
@@ -321,6 +318,70 @@ async void OnValuePropertyChanged(bool shouldUpdateSource)
 			Control.BorderColor = BorderColor;
 		}
 
+		async void OnSourcePropertyChanged(bool isBindingContextChanged)
+		{
+			if (Control == null)
+				return;
+
+			// If any thread is already waiting for its turn for updating the source
+			// then the current thread needn't have to stay in the queue in this case.
+			if (Interlocked.CompareExchange(ref isWaitingForSourceUpdateValue, 1, 0) == 1)
+				return;
+
+			try
+			{
+				if (!isBindingContextChanged)
+					imageLoadingTokenSource?.Cancel();
+
+				// Only one thread can update the source at the same time.
+				await imageSourceSemaphore.WaitAsync();
+
+				Interlocked.Exchange(ref isWaitingForSourceUpdateValue, 0);
+
+				if (isBindingContextChanged && (sourceBindingContext?.Equals(BindingContext) ?? false))
+					return;
+
+				// Wait for possible BindingContext change to prevent double image loading.
+				await Task.Delay(1);
+
+				var source = Source;
+				if (source != null)
+					SetInheritedBindingContext(source, BindingContext);
+
+				sourceBindingContext = source?.BindingContext;
+
+				imageLoadingTokenSource = new CancellationTokenSource();
+
+				var streamTask = GetStreamImageLoadingTask(source, imageLoadingTokenSource.Token);
+				if (streamTask != null)
+				{
+					var stream = await streamTask;
+					Image.IsVisible = stream != null;
+					source = ImageSource.FromStream(() => stream);
```

Comment:
@AndreiMisiukevich , Just need to catch a possible TaskCanceledException.

```
					try
					{
						var stream = await streamTask;
						Image.IsVisible = stream != null;
						source = ImageSource.FromStream(() => stream);
					}
					catch (TaskCanceledException)
					{
					}
```

Reply:
Do you think it will fix the issue?
Maybe we should copy one stream to another?

Reply:
It will only prevent app from crashing due to `TaskCanceledException`.

I don't know why it's not working if it's inside a ListView with CachingStrategy="RetainElement" (which is the default value).

Reply:
done